UKHomeOffice/nmsw-ui
# NMSW-1275 & NMSW-1280 Add error holding pages for crew and passenger

# Ticket



----

## AC

GIVEN I have submitted a FAL5/6 AND one or more person receives a Time out error (0T)
WHEN I am on the holding page
THEN a message is displayed to the user that support is looking into it (And no people are displayed)

GIVEN I have submitted a FAL5/6
WHEN one or more person receives ERROR_USER
THEN display error message on holding page

We identify errors by checking for 'Error System' and 'Error User' in the API response
These pages need applying to both Crew and Passenger checks

----

## To test

- upload a large crew file
- > you should get the `Crew holding page`
- upload a tiny crew file
- > you should get the `Crew completed page`
- upload a large passenger file
- > you should get the `Passenger holding page`
- upload a tiny passenger file
- > you should get the `UPT results page`

### To force the System error page to show
Comment out the `stepStatus === DECLARATION_STEP_STATUS_CIPROCESSING` on the VoyageHoldingPage.jsx file
and change `DECLARATION_STEP_STATUS_ERRORSYSTEM` to `DECLARATION_STEP_STATUS_CIPROCESSING`

- upload a large crew file
- > you should see the system error message with a link to task details page
- upload a large passenger file
- > you should see the system error message with a link to task details page

### To force the User error page to show
Comment out the `stepStatus === DECLARATION_STEP_STATUS_CIPROCESSING` on the VoyageHoldingPage.jsx file
and change `DECLARATION_STEP_STATUS_ERRORUSER` to `DECLARATION_STEP_STATUS_CIPROCESSING`

- upload a large crew file
- > you should see the system error message with a button to the FAL5 upload page
- upload a large passenger file
- > you should see the system error message with a button to the FAL6 upload page
----

## Notes
The BE is not returning any error statuses at the moment so we can only test against fixtures / forcing status change in FE
